packages/ui/src/utils/numberHelpers.ts

```diff
@@ -0,0 +1,173 @@
+ * @returns The formatted abbreviated string
+ */
+export function formatNumberAsAbbr(
+  value: number | string,
+  opts: NumberAsAbbrOptions = {},
+  locale = "en-US",
+) {
+  let num = typeof value === "string" ? Number(value) : value;
+  let suffix = "";
+
+  if (Math.abs(num) < 1_000) {
+    suffix = "";
+  } else if (Math.abs(num) < 1_000_000) {
+    if (opts.removeKAbbr) {
+      return num;
+    } else {
+      suffix = "K";
+      num = num / 1_000;
+    }
+  } else if (Math.abs(num) < 1_000_000_000) {
+    suffix = "M";
+    num = num / 1_000_000;
+  } else if (Math.abs(num) < 1_000_000_000_000) {
+    suffix = "B";
+    num = num / 1_000_000_000;
+  }
+
+  // In the future we can extend this to use any currency symbol
+  const symbol = opts.hideSymbol ? "" : "$";
+
+  if (opts.decimalPlaces === undefined) {
+    // Use 2 decimals for fiat and 4 for non fiat
+    opts.decimalPlaces = symbol ? 2 : 4;
+  }
+
+  const localeString = num.toLocaleString(locale, {
+    minimumFractionDigits: opts.decimalPlaces,
+    maximumFractionDigits: opts.decimalPlaces,
+  });
+
+  // If the return string is -0.00 or some variant, strip the negative
+  if (localeString.match(/-0\.?[0]*$/)) {
+    return localeString.replace("-", "");
+  }
+
+  return `${symbol}${localeString}${suffix}`;
+}
+
+/**
+ * Checks if an amount is considered "dust" (negligible or close to zero).
+ * @param amount - The amount to check
+ * @param threshold - The threshold for dust amounts (default is 1e-10)
+ * @returns true if the amount is considered dust
+ */
+export function isDustAmount(
+  amount: number | string,
+  threshold: number = 1e-10,
+): boolean {
+  const amountNumber = typeof amount === "string" ? Number(amount) : amount;
+
+  return Math.abs(amountNumber) < threshold;
+}
+
+/**
+ * Parses a string value into a JSBI BigInt
+ * @param value - The string value to parse
+ * @param decimals - The number of decimal places in the value
+ * @returns The parsed JSBI BigInt
+ */
+export const parseToBigInt = (value: string, decimals: number): JSBI => {
+  const [integerPart, fractionalPart = ""] = value.split(".");
+  const fractionalPartPadded = fractionalPart
+    .padEnd(decimals, "0")
+    .slice(0, decimals);
+  const wholeNumberString = `${integerPart}${fractionalPartPadded}`;
+  return JSBI.BigInt(wholeNumberString);
+};
+
+/**
+ * Formats a raw balance amount into a human-readable decimal number with 2 decimal places.
+ * Converts from the smallest unit (e.g., wei, satoshi) to the standard unit (e.g., ETH, BTC)
+ * by dividing by the appropriate power of 10 based on the `decimals` parameter.
+ *
+ * @param rawBalance - The raw balance amount as a string, represented in the smallest unit
+ * @param [decimals=18] - Number of decimal places to shift by (defaults to 18 for ETH compatibility)
+ * @returns The formatted balance as a string
+ */
+export function formatBalance(rawBalance: string, decimals = 18): string {
```

Comment:
There are now two instances of this function. You need to refactor the existing usage of this function and remove the other definition of this function.